stable/jenkins/templates/jenkins-master-route.yaml

```diff
@@ -18,7 +18,8 @@ metadata:
 {{- end }}
   name: {{ template "jenkins.fullname" . }}
 spec:
-  host: {{ .Values.master.route.path }}
+  host: "{{ .Values.master.route.host }}"
```

Comment:
Why do we need a different value `host` can't we re-use the already existing `hostName`?

Reply:
I feel it would be rather confusing to use `master.ingress.hostName` as the route's host, when we are not using anything else from `master.ingress.*` in the route definition. Unless it was changed to `master.hostName`.

Reply:
Agree that it's not ideal. The issue I see is that the hostname is also used to determine the Jenkins URL which is configured in the config map. Here is the template code:

```
{{/*
Returns the Jenkins URL
*/}}
{{- define "jenkins.url" -}}
{{- if .Values.master.jenkinsUrl }}
  {{- .Values.master.jenkinsUrl }}
{{- else }}
  {{- if .Values.master.ingress.hostName }}
    {{- if .Values.master.ingress.tls }}
      {{- default "https" .Values.master.jenkinsUrlProtocol }}://{{ .Values.master.ingress.hostName }}{{ default "" .Values.master.jenkinsUriPrefix }}
    {{- else }}
      {{- default "http" .Values.master.jenkinsUrlProtocol }}://{{ .Values.master.ingress.hostName }}{{ default "" .Values.master.jenkinsUriPrefix }}
    {{- end }}
  {{- else }}
      {{- default "http" .Values.master.jenkinsUrlProtocol }}://{{ template "jenkins.fullname" . }}:{{.Values.master.servicePort}}{{ default "" .Values.master.jenkinsUriPrefix }}
  {{- end}}
{{- end}}
{{- end -}}
```

https://github.com/helm/charts/blob/master/stable/jenkins/templates/_helpers.tpl#L51-L68

So it won't be as easy as introducing a new flag...

Reply:
1. It's seems the same due diligence was not done during the code review when this route was added initially. I don't think that, by any measure, this PR makes that situation worse than it is currently. Notice that the route's host can already be modified by setting `master.route.path`, which is not taken into account whatsoever in the `jenkins.url` definition.

2. Even ignoring the route for a second, that logic for `jenkins.url` doesn't seem right. If I don't set the `master.jenkinsUrl` value and set `master.ingress.enabled` to false, the `jenkins.url` template might still use the value `master.ingress.hostName`.

3. The logic for `jenkins.url` can easily be changed to take into account the route values:
```
{{- define "jenkins.url" -}}
{{- if .Values.master.jenkinsUrl }}
  {{- .Values.master.jenkinsUrl }}
{{- else if and .Values.master.ingress.enabled .Values.master.ingress.hostName }}
  {{- if .Values.master.ingress.tls }}
    {{- default "https" .Values.master.jenkinsUrlProtocol }}://{{ .Values.master.ingress.hostName }}{{ default "" .Values.master.jenkinsUriPrefix }}
  {{- else }}
    {{- default "http" .Values.master.jenkinsUrlProtocol }}://{{ .Values.master.ingress.hostName }}{{ default "" .Values.master.jenkinsUriPrefix }}
  {{- end }}
{{- else if and .Values.master.route.enabled .Values.master.route.host }}
  {{- default "http" .Values.master.jenkinsUrlProtocol }}://{{ .Values.master.route.host }}{{ default "" .Values.master.jenkinsUriPrefix }}
{{- else }}
  {{- default "http" .Values.master.jenkinsUrlProtocol }}://{{ template "jenkins.fullname" . }}:{{.Values.master.servicePort}}{{ default "" .Values.master.jenkinsUriPrefix }}
{{- end}}
{{- end -}}
```